fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -44,12 +44,15 @@ class _Lagrangian:
     """
 
     def __init__(self, X, sensitive_features, y, estimator, constraints, B, opt_lambda=True,
-                 sample_weight_name='sample_weight'):
+                 sample_weight_name='sample_weight',
+                 control_features=None):
```

Comment:
I think that this is becoming an anti-pattern here. Maybe we can change the signature to:
```python
    def __init__(self, X, y, estimator, constraints, B, opt_lambda=True,
                 sample_weight_name='sample_weight',
                 **kwargs)
```